pkg/controller/resourcegraphdefinition/controller.go

```
@@ -154,26 +155,46 @@ func (r *ResourceGraphDefinitionReconciler) findRGDsForCRD(ctx context.Context,
 	}
 }
 
-func (r *ResourceGraphDefinitionReconciler) Reconcile(ctx context.Context, o *v1alpha1.ResourceGraphDefinition) (ctrl.Result, error) {
+func (r *ResourceGraphDefinitionReconciler) Reconcile(ctx context.Context, o *v1alpha1.ResourceGraphDefinition) (res ctrl.Result, err error) {
+	var topologicalOrder []string
+	var resourcesInformation []v1alpha1.ResourceInformation
+	// defer function will ensure any errors that occur during reconcile will eventually be
+	// patched to the resource
+	defer func() {
+		// if the updateStatus error is a k8s NotFound error we want to skip it.
+		// If not, we will merge it to the returned error
+		if patchErr := r.updateStatus(ctx, o, topologicalOrder, resourcesInformation); patchErr != nil {
+			if apierrors.IsNotFound(patchErr) {
+				// we don't want to return an error and requeue if 
+				// rgd is not found (deleted)
+				err = nil
```

ignore the error if resource no longer exists